osu.Game/Graphics/Containers/ScalingContainer.cs

```diff
@@ -154,6 +155,15 @@ private void updateSize()
 
         private class ScalingBackgroundScreen : BackgroundScreenDefault
         {
+            private Bindable<WorkingBeatmap> beatmap;
+
+            [BackgroundDependencyLoader]
+            private void load(IBindable<WorkingBeatmap> beatmap)
+            {
+                this.beatmap = (Bindable<WorkingBeatmap>)beatmap;
+                this.beatmap.ValueChanged += _ => Next();
```

Comment:
Since there is only one field usage, I suggest to replace it with `(Bindable<WorkingBeatmap>)beatmap.ValueChanged += _ => Next();`

Reply:
you shouldn't be casting in the first place. create a bound copy and bind to its event. the method here is 100% wrong and causes a memory leak.